### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
Upgrade this solution to build with VS 2017 and dotnet-cli 1.0.0-preview4.

cc @Eilon @muratg

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<Project>
+  <Import Project="..\version.props" />
+  <Import Project="..\.build\common.props" Condition="Exists('..\.build\common.props')" />
+
+  <PropertyGroup>
+    <RepositoryUrl>https://github.com/aspnet/PlatformAbstractions</RepositoryUrl>
+    <RepositoryType>git</RepositoryType>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)Key.snk</AssemblyOriginatorKeyFile>
```
Comment (muratg) [inline build/common.props]: `MSBuildThisFileDirectory` value has a trailing backspace?

Comment (natemcmaster) [inline build/common.props]: Yup, always. cref https://msdn.microsoft.com/en-us/library/ms164309.aspx

**Comment by natemcmaster** [comment]
:watch: on https://github.com/aspnet/Universe/pull/466 to merge

File: src/Microsoft.Extensions.CommandLineUtils/Microsoft.Extensions.CommandLineUtils.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <Description>Command-line parsing API. Commonly used types:
+Microsoft.Extensions.CommandLineUtils.CommandLineApplication
```
Comment (Eilon) [inline src/Microsoft.Extensions.CommandLineUtils/Microsoft.Extensions.CommandLineUtils.csproj]: Love the newlines! 😄 Do they end up showing up correctly in the package's nuspec?

Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/Microsoft.Extensions.CommandLineUtils.csproj]: Yup!

image

File: build/common.props [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<Project>
+  <Import Project="..\version.props" />
+  <Import Project="..\.build\common.props" Condition="Exists('..\.build\common.props')" />
+
+  <PropertyGroup>
+    <RepositoryUrl>https://github.com/aspnet/PlatformAbstractions</RepositoryUrl>
```
Comment (cesarbs) [inline build/common.props]: `https://github.com/aspnet/Common`

Comment (natemcmaster) [inline build/common.props]: yeah, I caught that later :) 

https://github.com/aspnet/Common/commit/27b47956ebf045ec84a8b51634ed9706bd6ea3a5#diff-9780bf7a72094f1ac4b023e438b548ba

**Pull request merged**